src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Paris.cs

```diff
@@ -64,12 +65,24 @@ private async Task<ResultWrapper<ForkchoiceUpdatedV1Result>> ForkchoiceUpdated(F
         }
     }
 
-    private async Task<ResultWrapper<PayloadStatusV1>> NewPayload(ExecutionPayload executionPayload, int version)
+    private async Task<ResultWrapper<PayloadStatusV1>> NewPayload(IExecutionPayloadParams executionPayloadParams, int version)
     {
-        if (!executionPayload.ValidateParams(_specProvider, version, out string? error))
+        ExecutionPayload executionPayload = executionPayloadParams.ExecutionPayload;
+
+        if (!executionPayload.ValidateFork(_specProvider))
+        {
+            if (_logger.IsWarn) _logger.Warn($"The payload is not supported by the current fork");
+            return ResultWrapper<PayloadStatusV1>.Fail("unsupported fork", ErrorCodes.UnsupportedFork);
```

Comment:
Open question, should we use `ErrorCodes.InvalidParams` if `version < 3`? Before version 3 there is no error code `UnsupportedFork`.

Reply:
It's mentioned in https://github.com/ethereum/execution-apis/pull/426 for V2, an attempt to move towards more descriptive error codes.

Reply:
yes, I think so